Classes/Tree/NodeRepository.php

```diff
@@ -172,7 +172,7 @@ protected function retrieveByNamespace($namespace, $respectEnableFields)
         /*
          * RespectEnableFields = FALSE means, that all records are selected INCLUDING the deleted records
          */
-        if (!$respectEnableFields) {
+        if (!$respectEnableFields    && !$this->treeContext->isIncludeDeleted()) {
```

Comment:
Hallo Daniel. Dieser If-Teil hat mich auch ziemlich verwirrt und könnte mE sogar ganz raus, ich bin aber nicht sicher ob das irgendwelche Nebenwirkungen haben kann.
Bisher war es so, dass `deleted=0` immer gesetzt wurde, wenn die enabledFields ignoriert werden. Das ist mE aber gar nicht nötig, da das deletedFlag nicht zu den enabledFields gehört. D. h. egal ob  enabledFields true oder false ist, gelöschte Einträge werden nie berücksichtigt. Möchte man die gelöschten auch haben, muss explizit includeDeleted gesetzt werden. Aber ich weiß nicht, ob das in Typo3 schon immer so war, deswegen habe ich die Abfrage drin gelassen.

Wenn ich die Abfrage ändere in if `(!$respectEnableFields || $this->treeContext->isIncludeDeleted())` würde dass bedeuten, dass nie gelöschte Einträge abgefragt werden. Genau die will ich in dem Fall ja aber haben.

Ich hoffe das war verständlich ... ansonsten am besten nochmal im Gespräch :)